test.py

```diff
@@ -296,45 +310,53 @@ class BoostTestSuite(UnitTestSuite):
 
     # A cache of individual test cases, for which we have called
     # --list_content. Static to share across all modes.
-    _case_cache: Dict[str, List[str]] = dict()
+    _case_cache: Dict[str, List[Optional[str]]] = dict()
 
     def __init__(self, path, cfg: dict, options: argparse.Namespace, mode) -> None:
         super().__init__(path, cfg, options, mode)
 
-    async def create_test(self, name: str, suite, args) -> None:
-        exe = os.path.join("build", suite.mode, "test", suite.name, name)
-        if not os.access(exe, os.X_OK):
-            print(palette.warn(f"Boost test executable {exe} not found."))
-            return
-        options = self.options
-        allows_compaction_groups = self.all_can_run_compaction_groups_except != None and name not in self.all_can_run_compaction_groups_except
-        if options.parallel_cases and (name not in self.no_parallel_cases):
-            fqname = os.path.join(self.mode, self.name, name)
-            if fqname not in self._case_cache:
-                process = await asyncio.create_subprocess_exec(
-                    exe, *['--list_content'],
-                    stderr=asyncio.subprocess.PIPE,
-                    stdout=asyncio.subprocess.PIPE,
-                    env=dict(os.environ,
-                             **{"ASAN_OPTIONS": "halt_on_error=0"}),
-                    preexec_fn=os.setsid,
-                )
-                _, stderr = await asyncio.wait_for(process.communicate(), options.timeout)
-
-                case_list = [case[:-1] for case in stderr.decode().splitlines() if case.endswith('*')]
-                self._case_cache[fqname] = case_list
-
-            case_list = self._case_cache[fqname]
-            if len(case_list) == 1:
-                test = BoostTest(self.next_id((name, self.suite_key)), name, suite, args, None, allows_compaction_groups)
-                self.tests.append(test)
+    async def _exe_list_cases(self, exe: str) -> List[Optional[str]]:
```

Comment:
Please add a docstring to this function. Describe what `exe` is and what the function does.
Does `exe` in the function name start for `execute`? Please don't abbreviate this